wp/plugin/wp.ml

```diff
@@ -47,48 +47,31 @@ let update_default_num_unroll (num_unroll : int option) : unit =
   | Some n -> Pre.num_unroll := n
   | None -> ()
 
-let has_inline_funcs (to_inline : string list) : bool =
-  not (List.is_empty to_inline)
 
-let find_inline_funcs (inline_funcs : string list) ~to_find:(to_find : Sub.t Seq.t)
-    ~to_check:(to_check : Sub.t Seq.t) : Sub.t Seq.t =
-  if List.is_empty inline_funcs then
-    to_find
-  else
-    inline_funcs
-    |> List.map ~f:(find_func_in_one_of ~to_find ~to_check)
-    |> List.concat
-    |> Seq.of_list
+let match_inline (to_inline : string option) (subs : (Sub.t Seq.t)) : Sub.t Seq.t =
+  match to_inline with
+  | None -> Seq.empty
+  | Some to_inline -> let inline_pat = Re.Posix.re to_inline |> Re.Posix.compile in
```

Comment:
Do you think it's a good idea to have a warning log in the case we couldn't match on anything?

Reply:
Smart.

Reply:
I do!